### ISSUE TITLE
Does semgrep patterns follow code in multiple files i.e. source to sink across multiple files ?

### ISSUE THREAD

**Issue by L3m0nb4tt3ry** [ajinabraham/njsscan#16]
Does semgrep rules/njsscan check for patterns across multiple files.

e.g. User input is not validated and instead of returning response, there is a call to createResponse function, in this case, does semgrep/njsscan has the ability to follow createResponse and check if validation is applied or not in the createResponse function and then report ?
`import { createResponse blah blah} from 'blah/blah';`
`import escapestringblah  from 'blah/blah';`

`export async function getSomething(req: something, res: Response) {`
` const result = retriveSomething(blah blah,data);`
`    if (result) {`
 `     try {`
`blah blah`
`call to function escapeSctringsOrMaybeNot(something); `
`call to function createResponse(data);`
Files
/blah/escapestring.js
`'code to escape strings' or May be not`

/blag/createResponse.js
`'code to escape strings' or May be not`
`code to create response`

 Basically from source to sink across multiple files

**Comment by ajinabraham** [comment]
Semgrep has limited taint analysis support in a single file context. it cannot perform taint analysis across files at this time.
